### PR TITLE
Trim non split assemblies from published `refs` folder.

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/dotnet/cli/issues/2772

This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2757)

### CODE REVIEW

File: scripts/dotnet-cli-build/CompileTargets.cs [hunk 1]
```diff
@@ -692,30 +693,37 @@ private static void ChangeEntryPointLibraryName(string depsFile, string newName)
                 deps = JObject.ReadFrom(reader);
             }
 
-            var target = deps["targets"][deps["runtimeTarget"]["name"].Value<string>()];
-            var library = target.Children<JProperty>().First();
-            var version = library.Name.Substring(library.Name.IndexOf('/') + 1);
-            if (newName == null)
+        //    var target = deps["targets"][deps["runtimeTarget"]["name"].Value<string>()];
```
Comment (anurse) [inline scripts/dotnet-cli-build/CompileTargets.cs]: Remove commented code

File: src/Microsoft.Extensions.DependencyModel/DependencyContextPaths.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        public DependencyContextPaths(string application, string sharedRuntime)
+        {
+            Application = application;
+            SharedRuntime = sharedRuntime;
+        }
+
+        private static DependencyContextPaths GetCurrent()
+        {
+#if NETSTANDARD1_5
+            var deps = AppContext.GetData(DepsFilesProperty);
```
Comment (anurse) [inline src/Microsoft.Extensions.DependencyModel/DependencyContextPaths.cs]: :+1:

**Comment by anurse** [comment]
:shipit:

**Comment by pakrym** [comment]
@dotnet-bot test OSX x64 Release Build please
@dotnet-bot test Windows_NT x64 Release Build please
@dotnet-bot test Windows_NT x86 Debug Build please

File: src/Microsoft.Extensions.DependencyModel/Resolution/AppBaseCompilationAssemblyResolver.cs [hunk 3]
```diff
@@ -81,9 +98,15 @@ public bool TryResolveAssemblyPaths(CompilationLibrary library, List<string> ass
...
                 if (!resolved)
                 {
+                    // throw in case when we are published app and nothing found
+                    // because we cannot rely on nuget package cache in this case
+                    if (isPublished)
+                    {
                     throw new InvalidOperationException(
                         $"Can not find assembly file {assemblyFile} at '{string.Join(",", directories)}'");
                 }
+                    return false;
```
Comment (weshaggard) [inline src/Microsoft.Extensions.DependencyModel/Resolution/AppBaseCompilationAssemblyResolver.cs]: brace indention.

**Pull request merged**